### PR TITLE
optimizer-fix-controls, explain-index-merge: add optimizer fix control 52869 for index merge

### PR DESCRIPTION
**Pull request by time-and-fate**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/issues/52869

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: explain-index-merge.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ EXPLAIN SELECT /*+ USE_INDEX_MERGE(t, idx_a, idx_b, idx_c) */ * FROM t WHERE a >
 >
 > - TiDB 的索引合并特性在 v5.4.0 及之后的版本默认开启，即 [`tidb_enable_index_merge`](/system-variables.md#tidb_enable_index_merge-从-v40-版本开始引入) 为 `ON`。
 > - 如果查询中使用了 SQL 优化器 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-)，无论 `tidb_enable_index_merge` 开关是否开启，都会强制使用索引合并特性。当过滤条件中有无法下推的表达式时，必须使用 Hint [`USE_INDEX_MERGE`](/optimizer-hints.md#use_index_merget1_name-idx1_name--idx2_name-) 才能开启索引合并。
-> - 如果查询有除了全表扫描以外的单索引扫描方式可以选择，优化器不会自动选择索引合并，只能通过 Hint 指定使用索引合并。
+> - 如果查询有除了全表扫描以外的单索引扫描方式可以选择，优化器不会自动选择索引合并，只能通过 Hint 指定使用索引合并。从 v8.1.0 开始，这个限制可以通过 [Optimizer Fix Control 52869](/optimizer-fix-controls.md#52869-从-v810-版本开始引入) 解除。
```
Comment (songrijie) [inline explain-index-merge.md:93]: 这个限制是不是有风险，推荐什么样的客户打开？如果有风险，这里要提一下，建议客户要充分测试。

**Review by AilinKid — COMMENTED** [review]
LGTM

File: explain-index-merge.md (hunk 1 above)
Comment (time-and-fate) [reply explain-index-merge.md:93]: > 解除此限制能让优化器在更多查询中自动选择索引合并，但在索引合并的执行效率实际更低时，也会带来误选索引合并的风险，因此建议针对具体场景进行充分测试后打开解除此限制。

加一句这个如何呢，以及是不是在这边和 fix control 两个页面都加上这个？

Comment (songrijie) [reply explain-index-merge.md:93]: > 解除此限制能让优化器在更多查询中自动选择索引合并，但也有可能忽略其他更好的执行计划，因此建议在解除此限制前针对实际场景进行充分测试，确保不会带来性能回退。
这样？ 直接点。

**Review by songrijie — APPROVED** [review]

File: explain-index-merge.md (hunk 1 above)
Comment (songrijie) [reply explain-index-merge.md:93]: 两边都加上吧

File: optimizer-fix-controls.md [hunk 2]
```diff
@@ -69,4 +69,11 @@ SET SESSION tidb_opt_fix_control = '44262:ON,44389:ON';
...
 - 此开关控制优化器进行启发式访问路径选择的阈值。当某个访问路径（如 `Index_A`）的估算行数远小于其他访问路径时（默认为 `1000` 倍），优化器会跳过代价比较直接选择 `Index_A`。
-- `0` 表示关闭此启发式访问路径选择策略。
\ No newline at end of file
+- `0` 表示关闭此启发式访问路径选择策略。
+
+### [`52869`](https://github.com/pingcap/tidb/issues/52869) <span class="version-mark">从 v8.1.0 版本开始引入</span>
+
+- 默认值：`OFF`
+- 可选值：`ON`、`OFF`
+- 如[用 EXPLAIN 查看索引合并的 SQL 执行计划](/explain-index-merge.md)中“注意”小节中所说，如果查询有除了全表扫描以外的单索引扫描方式可以选择，优化器不会自动选择索引合并。
```
Comment (hfxsd) [inline optimizer-fix-controls.md:78]:
```suggestion
- 如[用 EXPLAIN 查看索引合并的 SQL 执行计划](/explain-index-merge.md#示例)中“注意”小节中所说，如果查询有除了全表扫描以外的单索引扫描方式可以选择，优化器不会自动选择索引合并。
```

**Review by hfxsd — APPROVED** [review]

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**